pandas/core/indexes/range.py

```diff
@@ -192,10 +192,10 @@ def _validate_dtype(cls, dtype: Dtype | None) -> None:
 
     # --------------------------------------------------------------------
 
-    # error: Return type "Type[NumericIndex]" of "_constructor" incompatible with return
+    # error: Return type "Type[Index]" of "_constructor" incompatible with return
     # type "Type[RangeIndex]" in supertype "Index"
     @cache_readonly
-    def _constructor(self) -> type[NumericIndex]:  # type: ignore[override]
+    def _constructor(self) -> type[Index]:
         """return the class to use for construction"""
         return NumericIndex
```

Comment:
this is going to be updated to return Index before long?

Reply:
Yes, couldn't do it here because there is some issue with empty indexes (`Index([])` and `NumericIndex([])` are not the same).